service/storage_proxy.cc

```diff
@@ -4403,7 +4403,14 @@ future<result<>> storage_proxy::schedule_repair(locator::effective_replication_m
     if (diffs.empty()) {
         return make_ready_future<result<>>(bo::success());
     }
-    return mutate_internal(diffs | std::views::values | std::views::transform([ermp] (auto& v) { return read_repair_mutation{std::move(v), ermp}; }), cl, false, std::move(trace_state), std::move(permit));
+    return mutate_internal(
+            diffs |
+                    std::views::values |
+                    std::views::transform([ermp] (auto& v) { return read_repair_mutation{std::move(v), ermp}; }) |
+                    // mutate_internal() is asynchronous, so we have to materialize this view,
+                    // before `diffs` is gone.
+                    std::ranges::to<std::vector<read_repair_mutation>>()
+            , cl, false, std::move(trace_state), std::move(permit));
```

Comment:
test?

I would say, better to have the rare trace case materialize the range, but read repair is rare enough (and expensive enough) that it's better to keep it simple.